Reduced contrast on non-text characters in VIM on both light &  dark

RussBrown00 opened the pull request:
This PR reduces the contrast between non-text and regular character colors

### Dark Half After Change
<img width="529" alt="Screen Shot 2020-10-14 at 5 13 33 PM" src="https://user-images.githubusercontent.com/4053376/96046054-df2e9580-0e40-11eb-903a-5a9f3bad47f9.png">

### Light Half Afer Change
<img width="518" alt="Screen Shot 2020-10-14 at 5 14 26 PM" src="https://user-images.githubusercontent.com/4053376/96046060-e2c21c80-0e40-11eb-9021-9532b9a2f67e.png">

sonph commented:
Looks ok to me. Is it possible to use `comment_fg` instead? I'd prefer that we use a small set of colors and also that's consistent with `Whitespace` (line 125).

RussBrown00 commented:
> Looks ok to me. Is it possible to use `comment_fg` instead? I'd prefer that we use a small set of colors and also that's consistent with `Whitespace` (line 125).

The color comment works, I preferred something lighter than the comments personally, but if you want to keep it small I'll change it. I don't think the Whitespace colorspace actually does anything anymore. I haven't had it effect anything in years.

sonph commented:
[`hl-Whitespace`](https://github.com/neovim/neovim/blob/759a05407f3e75cc002c0950563388735e969a94/runtime/doc/syntax.txt#L5068) is a thing in neovim (but not vim).

Hmm actually after looking closely at your screenshots I have to agree that the lighter colors work better, especially on comment lines in the light theme. I can go with option 1. The difference isn't that pronounced in the dark theme, can we pick another lighter or darker color here? @kdungs if you have any opinions.

Also, `cterm = 237` is the same as cursor line. Can you pick a different values? Refer to [this 256 color table](https://en.wikipedia.org/wiki/ANSI_escape_code#8-bit). 252 (`#d0d0d0`) for light theme and maybe 239 (`#4e4e4e`) for dark theme are closest to the colors you picked 🤔

kdungs commented:
No strong opinions from my side as long as the whitespace markers are not the same as the text :)

RussBrown00 commented:
I've updated the PR, hopefully, those changes are closer to desire. I've also moved the NonText next to WhiteSpace and they both use the same color palatte.

sonph reviewed (approved):
Looks good to me, thanks!

RussBrown00 commented:
> Looks good to me, thanks!

NP. Thanks for the theme. I assume you'll merge -- I can't
